bokeh/models/sources.py

```diff
@@ -138,23 +137,6 @@ def _data_from_df(df):
             new_data["index"] = index.tolist()
         return new_data
 
-    @classmethod
-    @deprecated("Bokeh 0.9.3", "ColumnDataSource initializer")
-    def from_df(cls, data):
-        """ Create a ``dict`` of columns from a Pandas DataFrame,
-        suitable for creating a ColumnDataSource.
-
-        Args:
-            data (DataFrame) : data to convert
-
-        Returns:
-            dict(str, list)
-
-        """
-        import warnings
-        warnings.warn("Method deprecated in Bokeh 0.9.3")
-        return cls._data_from_df(data)
-
     def to_df(self):
```

Comment:
I don't entirely understand what you're referring to removing or not.

I did remove the from_df method entirely (made it a private method) in an earlier commit, but we decided that exposing the method was likely useful. We use it in a bokeh server example, odo uses it to support bokeh as a backend and I've used it to validate data to send as JSON to an AJAXDataSource (and think it'll be more common going forward).

The original naming and implementation of the method wasn't very good, so this commit renames it to "parse_df" (not a great name, but at least there is a helpful docstring now) and reimplements it.